owrx/source/afedri.py

```diff
@@ -0,0 +1,105 @@
+from owrx.source.soapy import SoapyConnectorSource, SoapyConnectorDeviceDescription
+from owrx.form.input import Input, CheckboxInput, NumberInput
+from owrx.form.input.device import RemoteInput, TextInput
+from owrx.form.input.validator import RangeValidator
+from owrx.form.input.converter import OptionalConverter
+from owrx.form.input.validator import RequiredValidator
+from typing import List
+
+
+AFEDRI_DEVICE_KEYS = ["rx_mode"]
+AFEDRI_PROFILE_KEYS = ["r820t_lna_agc", "r820t_mixer_agc"]
+
+
+class AfedriAddressPortInput(TextInput):
+    def __init__(self):
+        super().__init__(
+            "afedri_adress_port",
+            "Afedri IP and Port",
+            infotext="Afedri IP and port to connect to. Format = IP:Port",
+            converter=OptionalConverter(),
+            validator=RequiredValidator(),
+        )
+
+
+class AfedriSource(SoapyConnectorSource):
+    def getSoapySettingsMappings(self):
+        mappings = super().getSoapySettingsMappings()
+        mappings.update({x: x for x in AFEDRI_PROFILE_KEYS})
+        return mappings
+
+    def getEventNames(self):
+        return super().getEventNames() + ["remote", "afedri_adress_port"] + AFEDRI_DEVICE_KEYS
+
+    def getDriver(self):
+        return "afedri"
+
+    def buildSoapyDeviceParameters(self, parsed, values):
+        params = super().buildSoapyDeviceParameters(parsed, values)
+
+        # replace driver with sopayRemote
+        if "remote" in values:
+            params = [v for v in params if "driver" not in v]  # remove present driver
+            params += [{"driver": "remote", "remote:driver": self.getDriver(), "remote": values["remote"]}]
```

Comment:
soapy remote is a separate source type (see `soapy_remote.py`). It should not be necessary to introduce custom handling here.

Reply:
Yes, but how to wrap a driver by soapyRemote?

Reply:
It’s all about multichannel Afedri devices. SoapyRemote is a way how to take independent channels from single instance of SoapyAfedri driver and split them across multiple soapy_connector.

Reply:
i see what you're trying to do. what i'm trying to tell you is that both soapyremote integration and multi-channel support are both things that need to be worked on on a bigger scale:

- if you want to do integrate soapyremote directly with the sources, please start a separate pull request that solves this in a way that makes it available to all soapy-based sources. please also make sure you work this out correctly with the feature detection since soapyremote is currently optional.
- if you want to add multi-channel support, we should definitely sit down and work out the use cases beforehand. just adding a separate device for every channel is a very limiting pattern, and does not provide to be a good user experience. we will need a new concept for this.

so i would recommend not including these things in this PR for now.

Reply:
I've removed soapyremote.